### PR TITLE
vendor: bump Pebble to ad44a62e4d04

### PR DESCRIPTION
**Pull request by jbowens**
```
ad44a62e db: ensure HasPointAndRange returns false after SetOptions
4f3c702c db: clean up and optimize Iterator.{set,save}RangeKey
4406dabc sstable: fix bug with two-level indexes, block properties and iter bounds
e7847d68 docs: add benchmark annotation for data quality issue
1fdc5c05 vfs: add option to turn off sync-on-close in syncingFile
```

Release note: None

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -47,7 +47,7 @@ require (
 	github.com/cockroachdb/go-test-teamcity v0.0.0-20191211140407-cff980ad0a55
 	github.com/cockroachdb/gostdlib v1.13.0
 	github.com/cockroachdb/logtags v0.0.0-20211118104740-dabe8e521a4f
```
Comment (erikgrinaker) [inline go.mod:49]: LGTM, once CI failure is fixed.

**Review by erikgrinaker — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (erikgrinaker) [reply go.mod:49]: Oops, app misclick. Doesn't matter I suppose.

**Review by nicktrav — APPROVED** [review]
:lgtm: :shipit: 

> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/82429#-:-N3fZmeu0HNOUq_-0Xca:byzm7aw)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @jbowens)

**Comment by jbowens** [comment]
TFTRs!

bors r=nicktrav,erikgrinaker

**Pull request merged**